Make consumer sampler/stats gathering compatible across debian/alpine…

mensfeld opened the pull request:
Continuation of https://github.com/karafka/karafka-web/pull/542

chen-anders commented:
> @chen-anders is there a reason why you used fiddle instead of ffi? If no bigger reason I would like to replace it to keep the karafka APIs aligned in regards to using FFI over Fiddle (other places use FFI). I am happy doing that myself, just want to be sure there was no bigger reason I am not aware of of using Fiddle. We would have to define Fiddle in the gemspec deps as well.

Oh I thought fiddle was part of the ruby stdlib if I'm not mistaken.

mensfeld commented:
@chen-anders thank you for your answer. Fiddle is one of gems being moved out of the stdlib: https://rubygems.org/gems/fiddle

> fiddle was loaded from the standard library, but will no longer be part of the default gems starting from Ruby 3.5.0

I will move it to FFI since you are ok (not action needed on your side). Thank you.

mensfeld commented:
Hey @chen-anders just a followup - i by no means forgot about this PR. Just had some family matters and will be getting back to it this week.

mensfeld commented:
@chen-anders, I changed the code a bit and would appreciate it if you took a second look. All seems fine, but having the original author double check is great. I moved the code to FFI and optimized the proc stats fetching by skipping other pid files.

mensfeld commented:
@chen-anders hope you feel better. It was merged and is scheduled for the upcoming release. Thank you so much :pray:
